habitat-lab/habitat/tasks/rearrange/rearrange_sim.py

```diff
@@ -492,31 +495,22 @@ def safe_snap_point(self, pos: np.ndarray) -> np.ndarray:
         """
         snap_point can return nan which produces hard to catch errors.
         """
-        new_pos = self.pathfinder.snap_point(pos)
-        island_radius = self.pathfinder.island_radius(new_pos)
-
-        if np.isnan(new_pos[0]) or island_radius != self._max_island_size:
-            # The point is not valid or not in a different island. Find a
-            # different point nearby that is on a different island and is
-            # valid.
-            new_pos = self.pathfinder.get_random_navigable_point_near(
-                pos, 1.5, 1000
+        if self.ep_info.scene_id not in self.scene_id_2_island_id:
+            island_areas = list(
+                map(
+                    self.pathfinder.island_area,
+                    range(self.pathfinder.num_islands),
+                )
             )
-            island_radius = self.pathfinder.island_radius(new_pos)
-
-        if np.isnan(new_pos[0]) or island_radius != self._max_island_size:
-            # This is a last resort, take a navmesh vertex that is closest
-            use_verts = [
-                x
-                for s, x in zip(self._island_sizes, self._navmesh_vertices)
-                if s == self._max_island_size
+            largest_island_id = island_areas.index(max(island_areas))
```

Comment:
We already store `self._max_island_size`. Can you instead use that instead of having to create a new cache?